src/store/audio.ts

```diff
@@ -493,6 +556,49 @@ export const audioStore = {
       const query = draft.audioItems[audioKey].query!;
       query.accentPhrases[accentPhraseIndex].moras[moraIndex].pitch = pitch;
     }),
+    [SET_AUDIO_MORA_VOICING]: createCommandAction<
+      State,
+      {
+        audioKey: string;
+        accentPhraseIndex: number;
+        moraIndex: number;
+        pitch?: number;
+      }
+    >((draft, { audioKey, accentPhraseIndex, moraIndex, pitch }) => {
+      const query = draft.audioItems[audioKey].query!;
+      const vowel =
+        query.accentPhrases[accentPhraseIndex].moras[moraIndex].vowel;
+      if (["i", "I", "u", "U"].indexOf(vowel) == -1) {
+        return;
+      }
+      if (pitch === undefined) {
+        query.accentPhrases[accentPhraseIndex].moras[moraIndex].pitch = 0;
+        query.accentPhrases[accentPhraseIndex].moras[moraIndex].vowel =
+          vowel.toUpperCase();
+      } else {
+        // -1 means there's no record in pitch history, using adjacent moras
+        if (pitch == -1) {
+          // use pitch of the mora behind it
+          try {
```

Comment:
隣接モーラの`pitch`が0のときに挙動が変になるかもしれません。

こうするのはどうでしょう？

1. 前後に`pitch!=0`の`pitch`があれば、2つの平均を取る
2. 前だけに`pitch!=0`の`pitch`があれば、前のをコピー
3. 後だけに`pitch!=0`の`pitch`があれば、後のをコピー
4. それ以外のときは`pitch=5.5`

---

The behavior may be wrong when the `pitch` of the adjacent mora is 0.

How about this?

1. if there is a `pitch!=0` before or after, take the average of the two pitch
2. if there is a `pitch` with `pitch!=0` only in the before, copy the before
3. if there is a `pitch` with `pitch!=0` only in the after, copy the after. 4.
4. otherwise, `pitch=5.5`.